web/src/main/java/io/fabric8/launcher/web/endpoints/OpenShiftEndpoint.java

```diff
@@ -70,7 +72,20 @@
             Set<OpenShiftCluster> subscribedClusters = clusterRegistry.getSubscribedClusters(securityContext.getUserPrincipal());
             for (OpenShiftCluster cluster : subscribedClusters) {
                 futures.add(identityProvider.getIdentityAsync(authorization, cluster.getId())
-                                    .thenApply(identity -> new ClusterVerified(cluster, identity.isPresent())));
+                                    .thenApply(identity -> {
+                                        if (!identity.isPresent()) {
+                                            return new ClusterVerified(cluster, false);
+                                        }
+                                        final ImmutableParameters.Builder builder = ImmutableParameters.builder().cluster(cluster).identity(identity.get());
+                                        final OpenShiftService service = openShiftServiceFactory.create(builder.build());
+                                        OpenShiftUser user = null;
+                                        try {
+                                            user = service.getLoggedUser();
+                                        } catch (KubernetesClientException e) {
+                                            //means that we have an invalid token e.g. cluster got deprovisioned
```

Comment:
I guess it's easier to read written like this:
```java
try {
    service.getLoggedUser();
    return new ClusterVerified(cluster, true);
} catch (KubernetesClientException e) {
    //means that we have an invalid token e.g. cluster got deprovisioned
    return new ClusterVerified(cluster, false);
}
```

Reply:
I think it depends... updated it nevertheless

Reply:
@edewit maybe it was subjective, I meant that it was easier at least to me :-)